numpy/lib/_function_base_impl.pyi

```diff
@@ -303,24 +307,87 @@ def diff(
     append: ArrayLike = ...,
 ) -> NDArray[Any]: ...
 
-@overload
+@overload  # float scalar
 def interp(
-    x: _ArrayLikeFloat_co,
+    x: _FloatLike_co,
+    xp: _ArrayLikeFloat_co,
+    fp: _ArrayLikeFloat_co,
+    left: _FloatLike_co | None = None,
+    right: _FloatLike_co | None = None,
+    period: _FloatLike_co | None = None,
+) -> float64: ...
+@overload  # float array
+def interp(
+    x: NDArray[floating | integer | np.bool] | _NestedSequence[_FloatLike_co],
     xp: _ArrayLikeFloat_co,
     fp: _ArrayLikeFloat_co,
-    left: None | _FloatLike_co = ...,
-    right: None | _FloatLike_co = ...,
-    period: None | _FloatLike_co = ...,
+    left: _FloatLike_co | None = None,
```

Comment:
Curious about this usage of `None = None`.

Reply:
Two things changed here:

1. The `None` should always be the last type within a union (according to the typing spec style guide, ruff, and `flak8-pyi`), so I tend to correct that in the places where I change things in the stubs (not the best for the diffs, but I can't really help myself 😅).
2. You noted before that annotaitons are a form of documentation. And when documenting functions, you also document the defaults. Using `= ...` in stubs was done by mypy's `stubgen` (for unknown reasons), and was therefore assumed to be the way to go. But for documentation, and also for things like IDE suggestions and type-checker error messages, it way more helpful if the actual default value is used, instead of a `...`. I believe the typing spec recently also started recommending doing this.